### PR TITLE
Use java 11 for dataflow flex templates

### PR DESCRIPTION
**Pull request by relud**
per https://github.com/mozilla/gcp-ingestion/issues/1534#issuecomment-893478454

also run beam and sink tests in both java 8 and 11 until we are fully off java 8

### CODE REVIEW

File: ingestion-beam/Dockerfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ RUN mvn package -Dmaven.test.skip=true
 # https://cloud.google.com/dataflow/docs/guides/templates/configuring-flex-templates#changing_the_base_image
 # and the versions available are defined here:
 # https://cloud.google.com/dataflow/docs/reference/flex-templates-base-images
-FROM gcr.io/dataflow-templates-base/java8-template-launcher-base:20210120_RC00
+FROM gcr.io/dataflow-templates-base/java11-template-launcher-base:20210120_RC00
```
Comment (whd) [inline ingestion-beam/Dockerfile:18]: Might as well move to 20210419_RC00 per the link in comment while we're doing this

**Review by jklukas — APPROVED** [review]
LGTM once we understand the failing test here.

**Comment by relud** [comment]
> LGTM once we understand the failing test here.

looks like we are hitting https://issues.apache.org/jira/browse/BEAM-6705

File: .circleci/config.yml [hunk 2]
```diff
@@ -308,6 +322,14 @@ workflows:
     - ingestion-beam-integration
     - ingestion-sink
     - ingestion-sink-integration
+    - ingestion-beam:
```
[the remark this answers is not in the document]
Comment (relud) [reply .circleci/config.yml]: switched to a build matrix and got `-java-11` and `-java-8`

File: ingestion-beam/src/main/java/com/mozilla/telemetry/metrics/KeyedCounter.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
  */
 public class KeyedCounter {
 
-  static final Map<String, Counter> counters = new HashMap<>();
+  static final Map<String, Counter> counters = new ConcurrentHashMap<>();
```
Comment (relud) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/metrics/KeyedCounter.java:14]: > LGTM once we understand the failing test here.

Concurrent access made it *appear* that the mapping function modified this map, and there is a change in HashMap.computeIfAbsent method specification for java 11:

> Throws:ConcurrentModificationException - if it is detected that the mapping function modified this map

Comment (jklukas) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/metrics/KeyedCounter.java:14]: This makes sense 👍

File: .circleci/config.yml [hunk 4]
```diff
@@ -304,11 +318,15 @@ workflows:
...
           tags:
             only: /.*/
-    - ingestion-beam
-    - ingestion-beam-integration
-    - ingestion-sink
-    - ingestion-sink-integration
+    - ingestion-beam: &java_matrix
+        matrix:
+          parameters:
+            executor: [java-8, java-11]
```
Comment (jklukas) [inline .circleci/config.yml:324]: This matrix solution is nice

**Review by jklukas — APPROVED** [review]

**Pull request merged**